Sources/Datadog/Core/Upload/DataUploadWorker.swift

```diff
@@ -90,3 +97,27 @@ internal class DataUploadWorker: DataUploadWorkerType {
         }
     }
 }
+
+extension DataUploadConditions.Blocker: CustomStringConvertible {
+    var description: String {
+        switch self {
+        case let .battery(level: level, state: state):
+            return "🔋 Battery: \(state) \(level)%"
+        case .lowPowerModeOn:
+            return "🔌 Low Power Mode: on"
+        case let .networkReachability(description: description):
+            return "📡 Reachability: " + description
+        }
+    }
+}
+
+extension DataUploadConditions.Report: CustomStringConvertible {
+    var description: String {
+        switch self {
+        case .go:
+            return "✅"
+        case let .noGo(blockers: blockers):
+            return "❌ → " + blockers.map { $0.description }.joined(separator: "; ")
```

Comment:
I think this may help more with self-diagnostic if this message is more explanatory giving "because" and "AND" context i.e.:

`🐶 → 19:00:10.949 [INFO] 💡 (logging) No upload. Batch to upload: NOT CHECKED, System conditions: ❌ [upload was skipped because: 🔋 Battery is unplugged (10%) AND 🔌 Low Power Mode is enabled AND 📡 Network reachability is: no]`

WDYT?

Reply:
This is clearer, I like it! I'll correct that.